### PR TITLE
feat(contact): added hubspot driver 

### PR DESCRIPTION
**Pull request by Nolan-Arnold**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

**Comment by damienwebdev** [comment]
@Nolan-Arnold this needs a rebase.

File: libs/contact/src/driver/hubspot/transformers/contact.transformer.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Injectable } from '@angular/core';
+import { DaffContactTransformerInterface } from '../../interfaces/contact-transformer.interface';
+import { DaffContactUnion } from '../../../models/contact-union';
+import { DaffHubspotRequest, DaffHubspotFormsTransformer } from '@daffodil/driver/hubspot';
+
+@Injectable({
```
Comment (damienwebdev) [inline libs/contact/src/driver/hubspot/transformers/contact.transformer.ts]: This should not be `providedIn` root.

File: libs/contact/src/driver/hubspot/models/config.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
```
Comment (damienwebdev) [inline libs/contact/src/driver/hubspot/models/config.ts]: This should not be injectable.

File: libs/contact/src/driver/hubspot/models/config.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+  providedIn: 'root'
+})
+export class DaffContactHubspotConfig {
+  portalId: string;
+  guid: string;
+  version? = '';
```
Comment (damienwebdev) [inline libs/contact/src/driver/hubspot/models/config.ts]: this should not likely exist.

File: libs/contact/src/driver/hubspot/transformers/json-builder.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import { FieldValue } from '../models/hubspot-response'
+
+
+export const jsonBuilder = (payload: object): FieldValue[] => {
```
Comment (damienwebdev) [inline libs/contact/src/driver/hubspot/transformers/json-builder.ts]: This code isn't used.

File: libs/contact/src/driver/hubspot/models/hubspot-response.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import { Time } from '@angular/common';
```
Comment (damienwebdev) [inline libs/contact/src/driver/hubspot/models/hubspot-response.ts]: This code isn't used.

File: libs/contact/integration-tests/hubspot.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      contactService.send(forumSubmission).subscribe();
+      const req = httpMock.expectOne(`${'https://api.hsforms.com/submissions/v3/integration/submit/123123/123123'}`);
+      expect(req.request.body).toEqual(
+        {
+          fields: [Object({ 'name': 'email', 'value': 'test@email.com' })],
+          context: Object({ hutk: null, pageUri: null, pageName: '' })
+        }
+      );
+      req.flush(mockReq);
+      httpMock.verify();
```
Comment (damienwebdev) [inline libs/contact/integration-tests/hubspot.service.spec.ts]: This duplicate `httpMock.verify` isn't necessary.

Comment (Nolan-Arnold) [reply libs/contact/integration-tests/hubspot.service.spec.ts]: All changes should be addressed in ac333252aa6378b2caf4f77f678fb62e65d91f08

File: libs/contact/src/index.ts [hunk 7]
```diff
@@ -1,3 +1,4 @@
 export { DaffContactServiceInterface } from './driver/interfaces/contact-service.interface'
 export { DaffContactUnion } from './models/contact-union';
-export { DaffContactDriver } from './driver/injection-tokens/contact-driver.token';
\ No newline at end of file
+export { DaffContactDriver } from './driver/injection-tokens/contact-driver.token';
+export { DaffContactTransformerInterface } from './driver/interfaces/contact-transformer.interface';
```
Comment (damienwebdev) [inline libs/contact/src/index.ts]: I think you need to expose the `DaffContactHubSpotDriverModule` right?

File: libs/contact/package.json [hunk 8]
```diff
@@ -14,6 +14,9 @@
     "test": "ng test contact --watch=false --browsers=ChromeHeadless",
     "publish": "(cd ../../dist/contact; npm publish --access=public)"
   },
+  "devDependencies": {
```
Comment (damienwebdev) [inline libs/contact/package.json]: You need a peerDependency on the `@daffodil/driver` too.

Comment (Nolan-Arnold) [reply libs/contact/package.json]: addressed in b84a1d943f378222f669d30fb8715392307091b4

**Pull request merged**